src/lib.rs

```diff
@@ -204,32 +202,38 @@ impl MediaType {
   }
 
   pub fn from_path(path: &Path) -> Self {
-    match path.extension() {
+    Self::from_path_like(path)
+  }
+
+  fn from_path_like(path: impl PathLike) -> Self {
+    match path.ext() {
       None => match path.file_name() {
         None => Self::Unknown,
-        Some(os_str) => {
-          let lowercase_str = os_str.to_str().map(|s| s.to_lowercase());
-          match lowercase_str.as_deref() {
-            Some(".tsbuildinfo") => Self::TsBuildInfo,
-            _ => Self::Unknown,
+        Some(file_name) => {
+          if file_name.eq_ignore_ascii_case(".tsbuildinfo") {
+            Self::TsBuildInfo
+          } else {
+            Self::Unknown
           }
         }
       },
-      Some(os_str) => {
-        let lowercase_str = os_str.to_str().map(|s| s.to_lowercase());
-        match lowercase_str.as_deref() {
-          Some("ts") => map_typescript_like(path, Self::TypeScript, Self::Dts),
-          Some("mts") => map_typescript_like(path, Self::Mts, Self::Dmts),
-          Some("cts") => map_typescript_like(path, Self::Cts, Self::Dcts),
-          Some("tsx") => Self::Tsx,
-          Some("js") => Self::JavaScript,
-          Some("jsx") => Self::Jsx,
-          Some("mjs") => Self::Mjs,
-          Some("cjs") => Self::Cjs,
-          Some("json") => Self::Json,
-          Some("wasm") => Self::Wasm,
-          Some("tsbuildinfo") => Self::TsBuildInfo,
-          Some("map") => Self::SourceMap,
+      Some(ext) => {
+        // using eq_ignore_ascii_case with if/elses seems to be ~40ns
+        // slower here, so continue to use to_lowercase()
+        let lowercase_str = ext.to_lowercase();
+        match lowercase_str.as_str() {
+          "ts" => map_typescript_like(path, Self::TypeScript, Self::Dts),
```

Comment:
Might be neat to have a macro that expands `"ts"` to `"ts" | "TS" | "Ts" | "tS"`... seems to reduce 268ns -> 220ns

Reply:
Maybe we could use https://doc.rust-lang.org/std/string/struct.String.html#method.eq_ignore_ascii_case instead? I think we could just create a macro that checks string length and uses that method for each case, and the compiler would be reasonably good at optimizing it.

Reply:
I tried that and it was slower than the match in this case for some reason (I think 40ns slower). It was faster for `tsbuildinfo` somewhere else in this file.